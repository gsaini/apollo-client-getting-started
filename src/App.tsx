import React from 'react';
import ExchangeRates from './components/ExchangeRates';
import logo from './logo.svg';
import './App.css';
import {
  Link
} from "react-router-dom";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <Link to="/books">Books</Link>
        <Link to="/todos">Todos</Link>

        <ExchangeRates />
      </header>
    </div>
  );
}

export default App;
