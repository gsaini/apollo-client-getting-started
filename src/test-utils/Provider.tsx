import React from "react";
import { act, render, RenderResult } from "@testing-library/react";
import { MockedProvider, MockedProviderProps } from '@apollo/client/testing';
import { cache } from '../apollo';


const customRender = async (
    childComponent: JSX.Element,
    options?: MockedProviderProps
): Promise<RenderResult> => {
    const { mocks = [], resolvers = {} } = options || {};
    const wrapper = render(<MockedProvider
        mocks={mocks}
        resolvers={resolvers}
        cache={cache}>
        {childComponent}
    </MockedProvider>)

    await act(() => new Promise((resolve) => setTimeout(resolve, 0)));

    return wrapper;
}

export { customRender as render };