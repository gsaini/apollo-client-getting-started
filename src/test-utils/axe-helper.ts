import { configureAxe } from "jest-axe";

export const axe = configureAxe({
  rules: {
    // for demonstration only, don't disable rules that need fixing.
    "image-alt": { enabled: false },
  },
});
