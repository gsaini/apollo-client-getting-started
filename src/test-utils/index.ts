export * from "./axe-helper";
export * from "./Provider";
