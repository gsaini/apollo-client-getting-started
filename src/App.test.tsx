import React from 'react';
import { render } from '@testing-library/react';
import { generateImage } from 'jsdom-screenshot';
import { axe } from './test-utils';

import App from './App';

describe('App', () => {
  it('should render snapshot', async () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('should render visual snapshot', async () => {
    render(<App />);

    const screenshot = await generateImage();
    expect(screenshot).toMatchImageSnapshot();
  });

  it('should validate accessibility voilations', async () => {
    const { container } = render(<App />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
