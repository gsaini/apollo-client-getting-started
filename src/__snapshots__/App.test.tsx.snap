// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p
        class="paragraph-cls"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Atque ex perferendis maxime voluptatem molestiae? Qui cupiditate asperiores ipsa nesciunt, iusto minima voluptatem alias sit porro sunt ad veritatis reiciendis? Deserunt.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;
